crates/aptos-api-tester/src/main.rs

```diff
@@ -0,0 +1,578 @@
+        block_height: U64(98764388),
+        block_hash: HashValue::from_str(
+            "f1fcfd799f88a2526d649d5a2cf227c14b77ef374b5daf6fb5d4987c430a91dd",
+        )
+        .context(ERROR_OTHER)?,
+        block_timestamp: U64(1688165648699603),
+        first_version: U64(565767992),
+        last_version: U64(565767994),
+        transactions: None,
+    };
+
+    ensure!(
+        actual_block.block_height == expected_block.block_height,
+        "{} expected {}, got {}",
+        FAIL_WRONG_BLOCK_HEIGHT,
+        expected_block.block_height,
+        actual_block.block_height,
+    );
+    ensure!(
+        actual_block.block_hash == expected_block.block_hash,
+        "{} expected {}, got {}",
+        FAIL_WRONG_BLOCK_HASH,
+        expected_block.block_hash,
+        actual_block.block_hash,
+    );
+    ensure!(
+        actual_block.block_timestamp == expected_block.block_timestamp,
+        "{} expected {}, got {}",
+        FAIL_WRONG_BLOCK_TIMESTAMP,
+        expected_block.block_timestamp,
+        actual_block.block_timestamp,
+    );
+    ensure!(
+        actual_block.first_version == expected_block.first_version,
+        "{} expected {}, got {}",
+        FAIL_WRONG_FIRST_VERSION,
+        expected_block.first_version,
+        actual_block.first_version,
+    );
+    ensure!(
+        actual_block.last_version == expected_block.last_version,
+        "{} expected {}, got {}",
+        FAIL_WRONG_LAST_VERSION,
+        expected_block.last_version,
+        actual_block.last_version,
+    );
+    ensure!(
+        actual_block.transactions == expected_block.transactions,
+        "{} expected {:?}, got {:?}",
+        FAIL_WRONG_TRANSACTIONS,
+        expected_block.transactions,
+        actual_block.transactions,
+    );
+
+    Ok(SUCCESS)
+}
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    match probe_getaccount_1().await {
+        Ok(result) => println!("{:?}", result),
+        Err(e) => println!("{:?}", e),
+    }
+    match probe_getaccount_2().await {
+        Ok(result) => println!("{:?}", result),
+        Err(e) => println!("{:?}", e),
+    }
+    match probe_getaccountbalance_1().await {
+        Ok(result) => println!("{:?}", result),
+        Err(e) => println!("{:?}", e),
+    }
+    match probe_getaccountbalance_2().await {
+        Ok(result) => println!("{:?}", result),
+        Err(e) => println!("{:?}", e),
+    }
+    match probe_getaccountbalanceatversion_1().await {
+        Ok(result) => println!("{:?}", result),
+        Err(e) => println!("{:?}", e),
+    }
+    match probe_getaccountbalanceatversion_2().await {
+        Ok(result) => println!("{:?}", result),
+        Err(e) => println!("{:?}", e),
+    }
+    match probe_getblockbyheight_1().await {
+        Ok(result) => println!("{:?}", result),
+        Err(e) => println!("{:?}", e),
+    }
+    match probe_getblockbyversion_1().await {
+        Ok(result) => println!("{:?}", result),
+        Err(e) => println!("{:?}", e),
+    }
```

Comment:
Let's teach some generics and DRY.

Something you can do here is to make a function that will take in the surrounding async function and handle it.

```
async fn save_results<
  Output: std::fmt::Debug,
  Fut: futures::Future<Output = anyhow::Result<Output>>
>(fut: Fut) {
    match future.await {
      Ok(result) => log_success(),
      Err(result) => log_failure()
    }
}
```

then you can simply call for each one:
```
save_results(probe_getblockbyversion_1()).await;
```

Additionally, it would be good to look into storing results in a structured way:
1. Did it succeed or fail?
2. Why did it succeed or fail?
3. How long did it take?
4. etc. etc.

This would allow for keeping track of it all and then possibly parse it / use something else to use the results after.